pkg/image/registry/image/strategy.go

```diff
@@ -44,6 +45,25 @@ func (imageStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
+func (imageStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (imageStrategy) PrepareForUpdate(obj, old runtime.Object) {
```

Comment:
No, it's the pull spec of the first place we saw the image. We store the
manifest in its entirety in DockerImageManifest.

On Thursday, September 17, 2015, Michal Minar <notifications@github.com>
wrote:

> In pkg/image/registry/image/strategy.go
> <https://github.com/openshift/origin/pull/4666#discussion_r39732078>:
>
> > @@ -44,6 +45,25 @@ func (imageStrategy) AllowCreateOnUpdate() bool {
> >  	return false
> >  }
> >
> > +func (imageStrategy) AllowUnconditionalUpdate() bool {
> > +	return false
> > +}
> > +
> > +// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
> > +func (imageStrategy) PrepareForUpdate(obj, old runtime.Object) {
>
> I thought this is actually a composition of Manifest.Name
> <https://github.com/docker/distribution/blob/master/manifest/schema1/manifest.go#L35>
> and Manifest.Tag
> <https://github.com/docker/distribution/blob/master/manifest/schema1/manifest.go#L38>.
> I believe that separation of name from manifest is on distribution's TODO (docker/distribution#62
> (comment)
> <https://github.com/docker/distribution/pull/62#issuecomment-73991387>).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4666/files#r39732078>.
>